app/models/user_stat.rb

```diff
@@ -151,12 +154,30 @@ def update_distinct_badge_count
   end
 
   # topic_reply_count is a count of posts in other users' topics
-  def update_topic_reply_count
-    self.topic_reply_count = Topic
-      .joins("INNER JOIN posts ON topics.id = posts.topic_id AND topics.user_id <> posts.user_id")
-      .where("posts.deleted_at IS NULL AND posts.user_id = ?", self.user_id)
-      .distinct
-      .count
+  def calc_topic_reply_count!(max, start_time = nil)
+    sql = <<~SQL
+    SELECT COUNT(*) count
+    FROM (
+      SELECT DISTINCT posts.topic_id
+      FROM posts
+      INNER JOIN topics ON topics.id = posts.topic_id
+      WHERE posts.user_id = ?
+      AND topics.user_id <> posts.user_id
+      AND posts.deleted_at IS NULL AND topics.deleted_at IS NULL
+      AND topics.archetype <> 'private_message'
```

Comment:
Looks like this is a new condition. Is it intended?

Reply:
TL3 requirements checked for `.public_posts` https://github.com/discourse/discourse/pull/9769/files#diff-71b275fc6f97d344de4978707d0f2fdeL147

Reply:
TL2 didn't as far as I know, but I guess it's better that way 👍